en/ros/ros2_comm.md

```diff
@@ -1,222 +1,292 @@
+In other words, it makes the executables that were just built available in the current terminal.
+
+In the same terminal:
+
+1. Source the `local_setup.bash` (you should also source `setup.bash` if in a new terminal).
 
    ```sh
-   RECEIVED DATA FROM SENSOR COMBINED
-   ================================
-   ts: 870938190
-   gyro_rad[0]: 0.00341645
-   gyro_rad[1]: 0.00626475
-   gyro_rad[2]: -0.000515705
-   gyro_integral_dt: 4739
-   accelerometer_timestamp_relative: 0
-   accelerometer_m_s2[0]: -0.273381
-   accelerometer_m_s2[1]: 0.0949186
-   accelerometer_m_s2[2]: -9.76044
-   accelerometer_integral_dt: 4739
+   source install/local_setup.bash
    ```
+1. Now launch the example.
+   Note here that we use `ros2 launch`, which is described below.
 
-You can also verify the rate of the message using `ros2 topic hz`.
-E.g. in the case of `sensor_combined` use `ros2 topic hz /fmu/sensor_combined/out`:
-   ```sh
-   average rate: 248.187
-   	min: 0.000s max: 0.012s std dev: 0.00147s window: 2724
-   average rate: 248.006
-   	min: 0.000s max: 0.012s std dev: 0.00147s window: 2972
-   average rate: 247.330
-   	min: 0.000s max: 0.012s std dev: 0.00148s window: 3212
-   average rate: 247.497
-   	min: 0.000s max: 0.012s std dev: 0.00149s window: 3464
-   average rate: 247.458
-   	min: 0.000s max: 0.012s std dev: 0.00149s window: 3712
-   average rate: 247.485
-   	min: 0.000s max: 0.012s std dev: 0.00148s window: 3960
    ```
+   ros2 launch px4_ros_com sensor_combined_listener.launch.py
+   ```
+
+If this is working you should see data being printed on the terminal/console where you launched the ROS listener:
+
+```sh
+RECEIVED DATA FROM SENSOR COMBINED
+================================
+ts: 870938190
+gyro_rad[0]: 0.00341645
+gyro_rad[1]: 0.00626475
+gyro_rad[2]: -0.000515705
+gyro_integral_dt: 4739
+accelerometer_timestamp_relative: 0
+accelerometer_m_s2[0]: -0.273381
+accelerometer_m_s2[1]: 0.0949186
+accelerometer_m_s2[2]: -9.76044
+accelerometer_integral_dt: 4739
+```
+
+## Controlling a Vehicle
+
+To control applications, ROS2 applications:
+
+- subscribe to (listen to) telemetry topics published by PX4
+- publish to topics that cause PX4 to perform some action.
+
+The topics that you can use are defined in [dds_topics.yaml](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/microdds_client/dds_topics.yaml), and you can get more information about their data in the [uORB Message Reference](../msg_docs/README.md).
+For example, [vehicle_global_position](../msg_docs/vehicle_global_position.md) can be used to get the vehicle global position, while [vehicle_command](../msg_docs/vehicle_command.md) can be used to command actions such as takeoff and land.
+
+The [ROS 2 Example applications](#ros-2-example-applications) examples below provide concrete examples of how to use these topics.
+
+## Compatibility Issues
+
+This section contains information that may affect how you write your ROS code.
+
+### ROS2 Subscriber QoS Settings
```

Comment:
@beniaminopozzan And this is the main qos settings topic. It isn't quite as good as yours, but it is more succinct and instructional - i.e. do this, it should just work. I ran out of puff - we can edit if you think it insufficient.

Reply:
As you said, most of the users will be satisfied by that, it get the job done!